document-generator-core/src/main/java/uk/gov/companieshouse/document/generator/core/utility/ApiResponseMapper.java

```diff
@@ -0,0 +1,24 @@
+package uk.gov.companieshouse.document.generator.core.utility;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.document.generator.core.service.response.ResponseObject;
+
+@Component
+public class ApiResponseMapper {
+
+    public ResponseEntity map(ResponseObject responseObject) {
+
+        switch(responseObject.getStatus()) {
+            case CREATED:
+                return ResponseEntity.status(HttpStatus.CREATED).body(responseObject.getData());
+            case NO_DATA_RETRIEVED:
+                return ResponseEntity.status(HttpStatus.BAD_REQUEST).build();
+            case DOCUMENT_NOT_RENDERED:
+                return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(responseObject.getData());
+               default:
```

Comment:
Can this be bought inline with the other case statements please?

Reply:
I'll do it on a future PR, as this has been approved and merged. before you commented. However, the default is only allowing me to place it in its current place in the IDE